src/CloudVolume.jl

```diff
@@ -17,78 +17,82 @@ export
 
 using PyCall
 @pyimport cloudvolume as cv
-const pyslice=pybuiltin(:slice)
+const pyslice = pybuiltin(:slice)
 
 function cached(f)
-  cache=Dict()
-  function my_f(args...; kwargs...)
-    if !haskey(cache, [args, kwargs])
-      cache[[args, kwargs]] = f(args...; kwargs...)
+    cache = Dict()
+    function my_f(args...; kwargs...)
+        if !haskey(cache, [args, kwargs])
+            cache[[args, kwargs]] = f(args...; kwargs...)
+        end
+        return cache[[args, kwargs]]
     end
-    return cache[[args, kwargs]]
-  end
 end
 
 CachedVolume = cached(cv.CloudVolume)
 CachedStorage = cached(cv.Storage)
 
 immutable CloudVolumeWrapper
-  val
-  function CloudVolumeWrapper(storage_string; mip=0,
-            bounded=true,
-            autocrop=false,
-            fill_missing=false,
-            cache=false, 
-            cdn_cache=false,
-            progress=false,
-            info=nothing,
-            provenance=nothing,
-            compress=nothing)
-    return new(CachedVolume(storage_string, mip=mip, 
-            bounded=bounded,
-            autocrop=autocrop,
-            fill_missing=fill_missing,
-            cache=cache,
-            cdn_cache=cdn_cache,
-            progress=progress,
-            info=info,
-            provenance=provenance,
-            compress=compress))
-  end
-end
-
-function Base.getindex(x::CloudVolumeWrapper, slicex::UnitRange, 
+    val::PyObject
+    function CloudVolumeWrapper(storage_string::AbstractString;
+            mip::Integer = 0,
+            bounded::Bool = true,
+            autocrop::Bool = false,
+            fill_missing::Bool = false,
+            cache::Union{Bool,AbstractString} = false,
+            cdn_cache::Union{Bool,Integer,AbstractString} = false,
+            progress::Bool = false,
+            info::Union{Dict,Void} = nothing,
+            provenance::Union{Dict,Void} = nothing,
+            compress::Union{Bool,AbstractString,Void} = nothing,
+            non_aligned_writes::Bool = false,
+            parallel::Union{Bool,Integer} = false,
+            output_to_shared_memory::Bool = false)
```

Comment:
output to shared memory should be Union{Bool,AbstractString}

Reply:
Thanks, fixed.
Also added a call to `unlink_shared_memory` to the `getindex()` functions. We are creating a copy there anyway; might as well clean up immediately.